packages/mermaid/src/diagrams/xychart/chartBuilder/TextDimensionCalculator.ts

```diff
@@ -0,0 +1,68 @@
+import { Dimension } from './Interfaces.js';
+
+export interface ITextDimensionCalculator {
+  getDimension(texts: string[], fontSize: number, fontFamily?: string): Dimension;
+}
+
+export class TextDimensionCalculator implements ITextDimensionCalculator {
+  getDimension(texts: string[], fontSize: number): Dimension {
+    return {
+      width: texts.reduce((acc, cur) => Math.max(cur.length, acc), 0) * fontSize,
+      height: fontSize,
+    };
+  }
+}
+
+export class TextDimensionCalculatorWithFont implements ITextDimensionCalculator {
+  private container: HTMLSpanElement | null = null;
+  private hiddenElementId = 'mermaid-text-dimension-calculator';
+  constructor(fontFamily?: string) {
+    if (document) {
+      let parentContainer = document.getElementById(this.hiddenElementId);
+      if (!parentContainer) {
+        parentContainer = document.createElement('div');
```

Comment:
This means we're polluting the main DOM with a hidden div. 
Although we use transient divs to calculate sizes, those are cleaned up afterwards. 
We need to remove this after the calculation. 

Also, if the mermaid rendering div has other properties applied, it won't reflect here. It might cause inconsistencies. 

Check the `computeWidthOfText` that we already use, and see if it can be reused.

Reply:
Since we required the function before we start rendering, so I have updated the function according to your suggestion, please check.

Reply:
Resolving this, because Sid had approved PR